### DISCUSSION TITLE
WSL 终端无内容显示无法操作（关闭开启 WSL 解决）

### DISCUSSION THREAD

**Discussion by SydenySun** [physics-data/faq#66] [Show and tell] [answered]
**情况描述：**
打开Windows Terminal之后发现无Ubuntu配置文件，尝试打开Ubuntu发现无内容显示，只有光标闪烁，如下图所示。
<div align=center>
<img width="400" alt="b65b4cee4b29947657d6eb7d28a9358" src="https://user-images.githubusercontent.com/55476681/130561922-2ba4f7a1-d5c4-4088-872c-81239271e6a9.png">
</div>

在命令行中输入where wsl及wsl --update --status命令，以及在PowerShell中输入wsl --help均可正常反应，但对于wsl --status，wsl --list，wsl --shutdown均无反应。

输入sc query lxssmanager,输出如下：
<div align=center>
<img width="400" alt="3a2ef97f0ec3f0979eb7c290558da2a" src="https://user-images.githubusercontent.com/55476681/130562685-7c29a2ca-196b-49e7-8952-b8c2f4046849.png">
</div>

**解决方法：**
在启动或关闭Windows功能中，关掉适用于Linux的Windows子系统，再重新打开。重启后一切恢复正常。

**尝试解决过程中的一些经验教训：**

不要尝试在Windows环境下修改/bin/文件夹下的文件，在网络搜索的教程指导下，我删除了bash文件，然而并没有解决问题，也无法重新安装bash。

**Comment by Harry-Chen** [comment, marked as answer]
不要尝试在Windows环境下修改任何 WSL 的文件（除了 `/mnt` 目录下）。

**Reply by zhangjing-GitHub-Code** [reply]
我之前用WSL好好的（1、2都能用），然后装了几个手机模拟器（夜神、蓝叠、雷电什么的），一开始下的（非雷电）打不开，然后哪个模拟器好像提示修复，重启后至少雷电能用了，然后WSL无了（提示BIOS开启虚拟化），最近将手机模拟器卸载了并`bcdedit /set hypervisorlaunchtype auto`了，结果重启后WSL命令直接无响应，过一会自动退出且无任何输出，这个（重新开启WSL）能管用吗？

**Reply by Harry-Chen** [reply]
有可能是模拟器损坏了虚拟机相关文件或者配置，可以考虑关闭 WSL 功能重新启用

**Comment by heroxbd** [comment]
不能盲从网络上的教程，它们的作者可能是小学生。

**Comment by carlos-lnmp** [comment]
这个问题我也遇到了。问题是这样产生的：我在window terminal 执行 wsl --update 命令之后 再执行 wsl --status 命令就没有反应了，Ubuntu 和 Debian 打开都没有反应。
解决方法是：重启电脑，就可以了

**Comment by Bigbeea** [comment]
http://t.csdnimg.cn/cMeN7  可以参考这个博客

**Comment by youhuiji** [comment]
> **情况描述：** 打开Windows Terminal之后发现无Ubuntu配置文件，尝试打开Ubuntu发现无内容显示，只有光标闪烁，如下图所示。
> 
> <img alt="b65b4cee4b29947657d6eb7d28a9358" width="400" src="https://user-images.githubusercontent.com/55476681/130561922-2ba4f7a1-d5c4-4088-872c-81239271e6a9.png">
> 在命令行中输入where wsl及wsl --update --status命令，以及在PowerShell中输入wsl --help均可正常反应，但对于wsl --status，wsl --list，wsl --shutdown均无反应。
> 
> 输入sc query lxssmanager,输出如下：
> 
> <img alt="3a2ef97f0ec3f0979eb7c290558da2a" width="400" src="https://user-images.githubusercontent.com/55476681/130562685-7c29a2ca-196b-49e7-8952-b8c2f4046849.png">
> **解决方法：** 在启动或关闭Windows功能中，关掉适用于Linux的Windows子系统，再重新打开。重启后一切恢复正常。
> 
> **尝试解决过程中的一些经验教训：**
> 
> 不要尝试在Windows环境下修改/bin/文件夹下的文件，在网络搜索的教程指导下，我删除了bash文件，然而并没有解决问题，也无法重新安装bash。

有用 感谢博主

**Reply by chopinone** [reply]
通过这个方法解决了，但是不一会又会出现这种情况。后来琢磨发现是和我的docker冲突了，我的电脑里面安装了docker，需要使用wsl -s Ubuntu 更改发行版为Ubuntu，改了之后我的wsl闪退问题也解决了

**Comment by heroxbd** [comment]
谢谢大家的后续讨论。
